macosx/Controller.mm

```diff
@@ -2368,16 +2368,20 @@ - (void)updateUI
     BOOL anyCompleted = NO;
     BOOL anyActive = NO;
 
-    for (Torrent* torrent in self.fTorrents)
     {
-        [torrent update];
+        // avoid having to wait for the same lock multiple times in the same operation
+        auto const lock = tr_sessionLock(self.sessionHandle);
+        for (Torrent* torrent in self.fTorrents)
+        {
+            [torrent update];
```

Comment:
I haven't inspected the code in depth, so this is more of a question than a change request -- how much risk is there in calling `[torrent update]` inside of the lock? it looks like update does enqueue a notification. Is that dispatched immediately, or later when the event loop is idle? (ie after we've released the lock)

Reply:
One of the macOS contributors will have to help us confirm this. I'm not joking when I say I've got obj-c++ literary of a kneecap. 🥶 

Tried my best to read the [docs for NSNotificationQueue](https://developer.apple.com/documentation/foundation/notificationqueue?language=objc). Sounds like there is an event loop similar to the core's libevent loop, and the notification will be dispatched after this `updateUI` invocation?

> Whereas a notification center distributes notifications when posted, notifications placed into the queue can be delayed until the end of the current pass through the run loop or until the run loop is idle.

CC @Coeur @nevack

Reply:
The notification enqueuing here is very safe: it's `NSNotificationCoalescingOnName` which means that if you send 1 thousand of them at the same time, it will only produce a small number of notifications. I don't know exactly how many, but I suspect around 30/sec or something adjusted for the refresh rate of the display.

In other words, you quickly enqueue 1000 times the same notification within 0.1 sec, it will only be posted 1-3 times.

Reply:
It's posted async, potentially during the lifetime of the lock.